source/Halibut/Transport/Proxy/HttpProxyClient.cs

```diff
@@ -287,6 +332,42 @@ void SendConnectionCommand(string host, int port)
             if (_respCode != HttpResponseCodes.OK)
                 HandleProxyCommandError(host, port);
         }
+        
+        async Task SendConnectionCommandAsync(string host, int port, CancellationToken cancellationToken)
+        {
+            var stream = TcpClient.GetStream();
+            var connectCmd = GetConnectCmd(host, port);
+            var request = Encoding.ASCII.GetBytes(connectCmd);
+
+            // send the connect request
+            await stream.WriteAsync(request, 0, request.Length, cancellationToken);
```

Comment:
Does this stream need to be wrapped with the yet-to-merge `TimeoutStream` so it can timeout and consistent cancellation?

Reply:
Yes does do we have a standard TODO for following this up?

I guess perhaps we should just never call `GetStream` and instead have some sort of `GetStreamWithTimeoutSupport` extension method and replace all `GetStream` calls with that.

Reply:
Sounds good, I have a standard comment of  `// TODO - ASYNC ME UP!` + details if relavent